integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala

```diff
@@ -442,7 +456,15 @@ class CarbonGlobalDictionaryGenerateRDD(
               model.hdfsLocation,
               dictionaryForDistinctValueLookUp,
               distinctValues)
-            sortIndexWriteTask.execute()
+            breakable {
+              if (!FileUtils
+                .validateTableExists(model.dictfolderPath + CarbonCommonConstants.FILE_SEPARATOR +
+                                     model.table.getTableId)) {
+                LOGGER.error(s"Table does not exists: ${ model.table.getTableUniqueName }")
+                break()
+              }
+              sortIndexWriteTask.execute()
+            }
```

Comment:
Remove file exist check from the executor code as it is a costly operation to check for file existence through RPC call to namenode for each column multiple times. Instead add this check in driver after global dictionary generation and delete the dictionary files created in case the table has been deleted.